go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -280,6 +280,160 @@ func (c *Controller) reconcile(name string) error {
 	return c.updateRAStatus(ra, hadUpdates, err)
 }
 
+func subnetsOverlap(subnet1, subnet2 string) (bool, error) {
+	ipNets, err := util.ParseIPNets([]string{subnet1, subnet2})
+	if err != nil {
+		return false, err
+	}
+
+	// Two subnets overlap if one contains the other's IP or vice versa
+	return ipNets[0].Contains(ipNets[1].IP) || ipNets[1].Contains(ipNets[0].IP), nil
+}
+
+// checkSubnetOverlaps validates that subnets from the current RA don't overlap
+// with any existing subnets in the same VRFs, both within the RA itself and
+// across different RAs
+func (c *Controller) checkSubnetOverlaps(ra *ratypes.RouteAdvertisements, currentRASubnets []string) error {
+	// For now, we only check when not using "auto" mode (will handle "auto" later)
+	if ra.Spec.TargetVRF == "auto" {
+		return nil
+	}
+
+	targetVRF := ra.Spec.TargetVRF
+
+	// Check for overlaps within the current RA
+	if err := checkWithinRAOverlaps(ra.Name, targetVRF, currentRASubnets); err != nil {
+		return err
+	}
+
+	// Check for overlaps between current RA and other RAs in the same VRF
+	if err := c.checkCrossRAOverlaps(ra, targetVRF, currentRASubnets); err != nil {
+		return err
+	}
```

Comment:
This do not make much sense the received RA is another RA we don't really care we just need a function where we pass the VRF and chec there is no overlapping, the created RA is just an event that state the VRF we have to check now.